x/oracle/keeper/tally_test.go

```diff
@@ -0,0 +1,201 @@
+			},
+			ExpectedWeightedMedian: sdk.MustNewDecFromStr("1.0"),
+			ExpectedValidatorClaimMap: map[string]types.Claim{
+				val(1).String(): types.NewClaim(10000, 10000, 1, val(1)),
+				val(2).String(): types.NewClaim(10000, 10000, 1, val(2)),
+			},
+		},
+		{
+			Description: "basic test for >50%% attack",
+			// Tally 5 votes
+			// rewardBand: 1.000000000000000000
+			// weightedMedian: 999.000000000000000000
+			// standardDeviation: 892.638336617916046807
+			// rewardSpread: 892.638336617916046807
+
+			Ballot: types.ExchangeRateBallot{
+				types.NewVoteForTally(sdk.MustNewDecFromStr("1.0"), "AAA", val(1), 10000),
+				types.NewVoteForTally(sdk.MustNewDecFromStr("1.0"), "AAA", val(2), 10000),
+				types.NewVoteForTally(sdk.MustNewDecFromStr("1.0"), "AAA", val(3), 10000),
+				types.NewVoteForTally(sdk.MustNewDecFromStr("1.0"), "AAA", val(4), 10000),
+				types.NewVoteForTally(sdk.MustNewDecFromStr("999.0"), "AAA", val(5), 40002),
+			},
+			RewardBand: sdk.MustNewDecFromStr("1.0"),
+			ValidatorClaimMap: map[string]types.Claim{
+				val(1).String(): types.NewClaim(10000, 0, 0, val(1)),
+				val(2).String(): types.NewClaim(10000, 0, 0, val(2)),
+				val(3).String(): types.NewClaim(10000, 0, 0, val(3)),
+				val(4).String(): types.NewClaim(10000, 0, 0, val(4)),
+				val(5).String(): types.NewClaim(40002, 0, 0, val(5)),
+			},
+			ExpectedWeightedMedian: sdk.MustNewDecFromStr("999.0"),
+			ExpectedValidatorClaimMap: map[string]types.Claim{
+				val(1).String(): types.NewClaim(10000, 0, 0, val(1)),
+				val(2).String(): types.NewClaim(10000, 0, 0, val(2)),
+				val(3).String(): types.NewClaim(10000, 0, 0, val(3)),
+				val(4).String(): types.NewClaim(10000, 0, 0, val(4)),
+				val(5).String(): types.NewClaim(40002, 40002, 1, val(5)),
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Log("TestTally Case:", testCase.Description)
+
+		resultValidatorClaimMap := copyValidatorClaimMap(testCase.ValidatorClaimMap)
+		weightedMedian, err := keeper.Tally(
+			testCase.Ballot,
+			testCase.RewardBand,
+			resultValidatorClaimMap,
+		)
+
+		s.Require().NoError(err)
+		s.Require().EqualValues(testCase.ExpectedWeightedMedian, weightedMedian)
+		s.Require().EqualValues(testCase.ExpectedValidatorClaimMap, resultValidatorClaimMap)
+	}
+}
+
+type tallyTestCase struct {
+	Description string
+
+	Ballot            types.ExchangeRateBallot
+	RewardBand        sdk.Dec
+	ValidatorClaimMap map[string]types.Claim
+
+	ExpectedValidatorClaimMap map[string]types.Claim
+	ExpectedWeightedMedian    sdk.Dec
+}
+
+func copyValidatorClaimMap(m map[string]types.Claim) map[string]types.Claim {
+	mm := make(map[string]types.Claim, len(m))
+	for k, v := range m {
+		mm[k] = v
+	}
+
+	return mm
+}
+
+func val(n int) sdk.ValAddress {
+	return sdk.ValAddress(fmt.Sprintf("val%08d__________", n))
+}
+
+// TestsBuildClaimsMapAndTally is a test for collection clams map and tallying.
+func (s *KeeperTestSuite) TestsBuildClaimsMapAndTally() {
```

Comment:
`TestsBuildClaimsMapAndTally` should be `TestBuildClaimsMapAndTally`

Reply:
Done.